hw/ip_templates/otp_ctrl/doc/theory_of_operation.md

```diff
@@ -115,6 +118,33 @@ The contents must go through periodic integrity checks and therefore the stored
 Note however that only the life cycle controller has access to this partition, i.e., the Direct Access Interface (DAI) cannot read nor write from/to the life cycle partition.
 
+## Zeroizing a Partition
+
+When a partition is configured as zeroizable, software can initiate a zeroization flow by zeroizing each word in a partition individually through the Direct-Access Interface (DAI).
+In order to efficiently identify already (or in the process of being) zeroized partitions, they are equipped with a dedicated 64-bit zeroization marker field that is inserted at the end of every zeroizable partition (see [Zeroization Detection](#zeroization-detection)).
+These marker fields can be inspected in the memory as part of the [Programmer's Guide](programmers_guide.md#direct-access-memory-map).
+The zeroization flow needs to be supported by the attached OTP macro ideally with a dedicated command.
+Zeroizing a word in a partition is semantically different to an ordinary write request as potentially already programmed are fuses are altered and redundancy measures such as ECC need to be disabled.
+Furthermore, as software needs to verify for each word that it was successfully zeroized, the OTP macro has to return the zeroized word back to the OTP controller which will further process it before releasing it to software for inspection.
+
+Zeroizing a partition affects its behavior during initialization and with respect to consistency and integrity checks (see [Partition Checks](#partition-checks) and [Design Details](#design-details)) .
+
+Note that all partition types except the `LIFE_CYCLE` partition can be configured as zeroizable.
+
+### Zeroization Detection
+
+There are two distinct cases when the OTP controller performs a zeroization detection by counting the number of set bits in a word and comparing the resulting value against predefined thresholds that account for potentially stuck-at-0 bits.
+
+  1. _Initialization_: As explained in [Checks on Zeroized Partitions](#checks-on-zeroized-partitions) the periodic consistency and integrity checks need to be disabled for a zeroized partition, furthermore when reading out the content of buffered partitions, the ECC checks need to be disabled as well.
+  This means that the zeroization marker field has to be checked first before performing the remaining initialization steps.
+  A partition is said to be in a zeroized state if and only if the number of set bits in the indicator field is greater or equal the macro-specific value `ZeroizationValidBound`.
+  This bound should include a margin that accounts for defects where certain fuse bits are perpetually stuck at 0.
+  2. _During Zeroization_: For compliance with OCP L.O.C.K., software needs to verify whether a word has been successfully zeroized.
+  In order to avoid releasing scrambled data to software in case of an error, the DAI will release the number of set bits in a zeroized word, if this number is greater or equal to the `ZeroizationValidBound`.
```

Comment:
Apparently the count of ones is always returned, so delete the text after `in a zeroized word`?